src/gui/processing/models/qgsmodelgraphicitem.cpp

```diff
@@ -180,13 +185,32 @@ QgsModelDesignerSocketGraphicItem::QgsModelDesignerSocketGraphicItem( QgsModelCo
 
 void QgsModelDesignerSocketGraphicItem::paint( QPainter *painter, const QStyleOptionGraphicsItem *, QWidget * )
 {
-  painter->setPen( QPen() );
-  painter->setBrush( QBrush( QColor( 0, 0, 0, mHoverState ? 200 : 33 ), Qt::SolidPattern ) );
+  QColor outlineColor = socketColor();
+  QColor fillColor = QColor( outlineColor );
+
+  if ( isInput() )
+  {
+    fillColor.setAlpha( isDefaultParameterValue() ? 30 : 255 );
+  }
+  else
+  {
+    // outputs are always filled sockets
+    fillColor.setAlpha( 255 );
+  }
+
+  // Outline style
+  painter->setPen( QPen( outlineColor, mHoverState ? mSocketOutlineWidth * 2 : mSocketOutlineWidth ) );
+
+  // Fill style
+  painter->setBrush( QBrush( fillColor, Qt::SolidPattern ) );
 
   painter->setRenderHint( QPainter::Antialiasing );
 
-  constexpr float DISPLAY_SIZE = 3.2;
-  painter->drawEllipse( position(), DISPLAY_SIZE, DISPLAY_SIZE );
+  constexpr float DISPLAY_SIZE = 4;
+  float ellipseOffset = 0.4;
```

Comment:
```suggestion
  constexpr float ELLIPSE_OFFSET = 0.4;
```

Also add a comment on why this is here